language-server/modules/langserver-compiler/src/main/java/org/ballerinalang/langserver/compiler/LSCompilerUtil.java

```diff
@@ -242,6 +242,37 @@ public static String getSourceRoot(Path filePath) {
         return fileRoot != null ? fileRoot : parentPath.toString();
     }
 
+    /**
+     * Returns top-level module path of a given file path.
+     *
+     * @param filePath file path
+     * @return top-level module path
+     */
+    public static Path getCurrentModulePath(Path filePath) {
+        Path projectRoot = Paths.get(LSCompilerUtil.getSourceRoot(filePath));
+        Path currentModulePath = projectRoot;
+        Path prevSourceRoot = filePath.getParent();
+        try {
+            if (prevSourceRoot == null || Files.isSameFile(prevSourceRoot, projectRoot)) {
+                return currentModulePath;
+            }
+            while (true) {
+                Path newSourceRoot = prevSourceRoot.getParent();
+                currentModulePath = prevSourceRoot;
+                String newSourceRootStr = (newSourceRoot != null) ? newSourceRoot.toString() : "";
```

Comment:
We can remove this assignment and instead use only the null check in the if condition

Reply:
Isn't this [in-line null check](https://en.wikipedia.org/wiki/%3F%3a#Java) is more readable than writing four lines for the `null` check?
```java
String newSourceRootStr = "";
if (newSourceRoot != null) {
    newSourceRootStr  = newSourceRoot.toString();
}
```

Reply:
That is ok to have. What I was suggesting, in line 263, we check for newSourceRoot == null, which is done already. Either we can remove this empty string assignment or remove the null check in line 263